docs/protobuf.md

```diff
@@ -116,3 +130,56 @@ Inside the `pb` module you will find all generated Rust types and their implemen
   a valid variant, so the Rust code needs to deal with a lot of `Option`s. As we
   don't expect other parts of Anki to send invalid messages, using an `InvalidInput`
   error or `unwrap_or_default()` is usually fine.
+
+# Communication between languages
+
+Currently, Rust does not send message to Python or Ts. That is, the
```

Comment:
'send' is somewhat misleading here, since the request does not return until the call has completed and a return value is provided - I'd use things like "call" or "invoke" a method instead.

I would lede with something like:

Anki's codebase uses three layers:

1) The core Rust layer
2) The Python layer
3) The web frontend/TypeScript layer

Each layer only makes RPC calls to the layer below it. TypeScript code makes calls to the Python-level HTTP server. Python code makes calls to the underlying Rust layer. The Rust layer never makes RPC calls.

Reply:
RPC call is tautological, but pretty common in my experience. But my point was more that Python→Rust and web→Python are 'calls' in the sense that they block until they receive a reply (assuming the TS promise is awaited). As you know, web.eval(WithCallback) is more akin to injecting some text into the JavaScript console, and it's not possible for the Python code to block until the reply is received.